server/matchmaker/algorithm.py

```diff
@@ -52,13 +54,37 @@ def find(self) -> List[Match]:
                     search, preferred, search.quality_with(preferred),
                     search.match_threshold, preferred.match_threshold
                 )
-                if not search.matches_with(preferred):
-                    continue
 
                 self._propose(search, preferred)
 
+        self._forcefully_match_unmatched_newbies()
+
         return self._remove_duplicates()
 
+    def _forcefully_match_unmatched_newbies(self):
+        unmatched_newbies = [
+            search for search in self.searches 
+            if search.is_single_ladder_newbie()
+            and not search in self.matches
+        ] 
+
+        while 2 <= len(unmatched_newbies):
```

Comment:
Any reason not to do
```suggestion
        while len(unmatched_newbies) >= 2:
```
? I feel like this is more readable

Reply:
I once read a nice blog post on style suggesting to only use "<" and never ">" because it makes things somewhat natural to think about when reading (since they are always sorted from left to right increasing/non-decreasing).
But I agree that it looks weird if one isn't used to it and it's certainly not important here ;)

Reply:
Interesting. I don’t think I agree with that reasoning :P, unless maybe you establish it as some standard for the entire code base